controller/src/beerocks/master/db/db.cpp

```diff
@@ -4302,3 +4302,67 @@ bool db::remove_client_entry_and_update_counter(const std::string &entry_name)
 
     return false;
 }
+
+sMacAddr db::get_candidate_client_for_removal()
```

Comment:
this API is used by the DB for maintaining a seamless "add new client" on one hand - while keeping the "clients_persistent_db_max_size" limit on the other.

I agree this is pure logic - but since the db is the only one to use it and since the db encapsulates the managing of the size-limitation, I still think this should be part of the db class.
Moreover - the user **new** client configuration is always preferred over existing client configurations and will always be added even if removal of an entry from the DB needs to be performed to "make room" for it - so I don't want the user to handle it externally.

Reply:
See my later comment to the use of this functionality.
The db is responsible to remove-upon-adding-to-full, but the client is responsible to the select who to remove.
The db provides the mechanism, the rule, the action that will be taken - the "what", the client provides the "how".

Reply:
as discussed offline, I have decided to leave it as is at this stage - but will consider introducing this change if we really require more than one way of determining the candidate client to be removed.